sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -504,8 +499,6 @@ def query_items(
             response_hook=response_hook,
             **kwargs
         )
-        if response_hook:
```

Comment:
We can't remove this?

Reply:
hmmm yeah this ones tricky....

Technically removing this shouldn't result in any new raised errors, because any hook logic would have needed to accommodate both the Dict input and the Pageable input....
I'm trying to think if there would have been any specific scenario that might stop working..... in theory the whole point of the response_hook is to access the headers right? And they're still going to get that....

What happens if the query returns no results? Does that mean I will go from one call to my response_hook (albeit with incorrect headers), to zero calls with no headers? Or would I still get the hook called with the specified empty response?

Reply:
yep that's exactly right - the first call with the query_iterable and the wrong headers happens, and then the next one looks something like this:
![image](https://github.com/user-attachments/assets/1b043b7f-486f-44b8-b100-c28813db8d29)

and like you mentioned, either way anyone who had been using this would have had to have logic to deal with both scenarios